### PR TITLE
Added types for node-horseman

### PR DESCRIPTION
**Pull request by apratheek**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

### CODE REVIEW

File: types/node-horseman/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+
+/*~ If this module is a UMD module that exposes a global variable 'myLib' when
+ *~ loaded outside a module loader environment, declare that global here.
+ *~ Otherwise, delete this declaration.
+ */
+
+export = horseman;
+
+declare namespace horseman {
+    class horseman {}
```
Comment (mhegazy) [inline types/node-horseman/index.d.ts]: do not see this in the node package:

```sh
> npm install  node-horseman

+ node-horseman@3.3.0
added 9 packages in 1.963s

> node
> var n = require("node-horseman");
undefined
> n;
{ [Function: Horseman]
  registerAction: [Function: registerAction],
  TimeoutError: [Function: SubError] }
> n.horseman
undefined

```

**Review by mhegazy — CHANGES_REQUESTED** [review]

File: types/node-horseman/index.d.ts (hunk 1 above)
Comment (apratheek) [reply types/node-horseman/index.d.ts]: Hi, 

The package is used as follows:
```
const horseman = require('node-horseman');
let horse = new horseman({
    timeout: 10000
});
horse.on('error', (msg: any, trace: any) => { console.log(msg, trace); })
+.open("http://example.org/").waitForNextPage().open("https://google.com").waitForNextPage().html().then((pageContent: string) => {
+    console.log(pageContent);
+    horse.close();
+});
```

The package requires `PhantomJS` (npm install phantomjs). I think I may not have added this as a dependency. Is that the issue? This is my first package, so I'm out of my depth here. Any help would be appreciated.

Comment (mhegazy) [reply types/node-horseman/index.d.ts]: I do not think you need this namespace/inner class. your other class declaration for `horseman` should be sufficient.

Comment (apratheek) [reply types/node-horseman/index.d.ts]: Hi, I tried to remove the namespace and directly export the class. It kinda made the initialization complicated, plus, I couldn't really get it to run on Node. Using the current structure, it works, though. Just to let you know, I'm an amateur at Typescript, so I don't exactly know how classes and namespaces go together.

I tried to run the test on my end, and it works. Am I missing something here?

File: types/node-horseman/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6",
+            "dom"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictNullChecks": true,
+        "baseUrl": "../",
```
Comment (andy-ms) [inline types/node-horseman/tsconfig.json]: Also add `"strictFunctionTypes": true,`

File: types/node-horseman/node-horseman-tests.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import * as horseman from 'node-horseman';
```
Comment (andy-ms) [inline types/node-horseman/node-horseman-tests.ts]: `import horseman = require("node-horseman");`

File: types/node-horseman/index.d.ts (hunk 1 above)
Comment (andy-ms) [reply types/node-horseman/index.d.ts]: Just delete the 3 lines of `declare namespace horseman` entirely and the code should work assuming you change the import style as shown in my other comment.

File: types/node-horseman/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,250 @@
+// Type definitions for node-horseman 3.3
+// Project: https://github.com/johntitus/node-horseman#readme
+// Definitions by: Pratheek Adidela <https://github.com/apratheek>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.2
+
+/*~ If this module is a UMD module that exposes a global variable 'myLib' when
```
Comment (andy-ms) [inline types/node-horseman/index.d.ts]: Remove this comment

File: types/node-horseman/tsconfig.json (hunk 2 above)
Comment (apratheek) [reply types/node-horseman/tsconfig.json]: Hi, thank you so much for helping me out. I've made and pushed the changes as stated. Should I create a new PR? Kindly let me know how to proceed. Thanks a ton!

Comment (andy-ms) [reply types/node-horseman/tsconfig.json]: No, making the changes here is good.

**Comment by andy-ms** [comment]
Thanks!

**Comment by apratheek** [comment]
Thank you!

**Pull request merged**